syntax = "proto3";
package game;


// 登录响应 (Server -> Client, 连接成功后第一条消息)
message LoginResponse {
    int32 your_id = 1;     // 你的 ID
    float init_x = 2;      // 你的出生位置 X
    float init_y = 3;      // 你的出生位置 Y
}
// 1. 玩家输入 (客户端 -> 服务端)
// 我们简化一下，不用位掩码，直接用枚举，方便你调试
message PlayerInput {
    int32 player_id = 1;
    enum InputType {
        NONE = 0;
        UP = 1;
        DOWN = 2;
        LEFT = 3;
        RIGHT = 4;
        FIRE = 5;
    }
    InputType input = 2;
}

// 2. 单个玩家的状态 (服务端内部用)
message PlayerState {
    int32 id = 1;
    float x = 2;
    float y = 3;
    float angle = 4; // 后面可以做旋转
}

// 3. 游戏世界快照 (服务端 -> 客户端)
// 服务器每帧把这个发给所有客户端，客户端收到后根据这个画图
message GameSnapshot {
    repeated PlayerState players = 1;
}